test/base/test_batch.py

```diff
@@ -150,13 +150,15 @@ def test_batch_cat_and_stack_and_empty():
     assert np.allclose(b5.b.c, [2, 0])
     assert np.allclose(b5.b.d, [1, 0])
     data = Batch(a=[False, True],
-                 b={'c': [2., 'st'], 'd': [1, None], 'e': [2., float('nan')]},
+                 b={'c': np.array([2., 'st'], dtype=np.object),
+                    'd': [1, None],
+                    'e': [2., float('nan')]},
                  c=np.array([1, 3, 4], dtype=np.int),
                  t=torch.tensor([4, 5, 6, 7.]))
     data[-1] = Batch.empty(data[1])
     assert np.allclose(data.c, [1, 3, 0])
     assert np.allclose(data.a, [False, False])
-    assert list(data.b.c) == ['2.0', '']
+    assert list(data.b.c) == [2.0, None]
```

Comment:
Add a testcase using `Batch.empty_`

Reply:
Added in 1a35bea31cb6273056fcf17857fbd5ccf42d8434 .